src/timing.c

```diff
@@ -113,18 +135,124 @@ void jl_destroy_timing(void)
     jl_ptls_t ptls = jl_current_task->ptls;
     jl_timing_block_t *stack = ptls->timing_stack;
     while (stack) {
-        _jl_timing_block_destroy(stack);
+        jl_timing_block_end(stack);
         stack = stack->prev;
     }
 }
 
-jl_timing_block_t *jl_pop_timing_block(jl_timing_block_t *cur_block)
+static const int get_timing_subsystem(const char *subsystem) {
+    const char **match = (const char **)bsearch(
+        subsystem, jl_timing_subsystems, JL_TIMING_SUBSYSTEM_LAST,
+        sizeof(const char *), (int(*)(const void*, const void*))strcmp
+    );
+    if (!match)
+        return JL_TIMING_SUBSYSTEM_LAST;
+
+    return (int)(match - &jl_timing_subsystems[0]);
+}
+
+// This function is analogous to __itt_event_create but for the counts backend
+//
+// `event` is required to live forever
+JL_DLLEXPORT jl_timing_counts_event_t *jl_timing_counts_event_create(const char *event) {
```

Comment:
This should take both the name and strlen of the name.

Reply:
Is that so that it can accept non-null-terminated strings? (presumably from Julia)

Reply:
Yes, and also to avoid wasting time computing strlen in general.

Reply:
This (timing-internal) API is just for C strings at the moment, so I think I'm going to leave it as-is.

We can change the prototype later if the implementation is updated to accept non-null-terminated strings